tests/unit/all.js

```diff
@@ -148,12 +148,24 @@ describe('code snippet example', () => {
     }
     Foo.prototype.d = 4;
     Bar.prototype.f = 6;
-    const extend = (target, ...sources) => {
-      let source = [];
-      sources.forEach(src => {
-        source = source.concat([src, Object.getPrototypeOf(src)])
-      })
-      return Object.assign(target, ...source)
+
+    const extend = (...args) => {
```

Comment:
This code still can be optimized. You don't need 3rd loop, you can do `obj[key] = source[key];` in `for..in` loop and it's much faster.

Reply:
@ODudek Yeah, you are right.

I do optimize the function in https://github.com/hexojs/hexo/pull/3987, but I forget to update this PR then. I will update this PR to newest then.